department-of-veterans-affairs/vets-website
# Covid 19 vax unsubscribe

## Description
This PR provides a URL to allow a user to unsubscribe from the Covid Vaccine Keep Me Informed updates.  
Related vets-api PR: https://github.com/department-of-veterans-affairs/vets-api/pull/5570

## Testing done
Unit test in place to ensure `unsubscribe` page loads. 
Cypress tests for success and fail end to end testing with mocked backend
manual testing in review instance:

1. Authenticate and submit a new form for that user
2. Navigate to the form page and reload if needed 
3. Using DevTools locate the GET Request to `/covid_vaccine/v0/registration` (this is the call to retrieve any previously entered data for this user)
4. Extract the  `id` from the Response  
![image](https://user-images.githubusercontent.com/2481110/104351640-a64f9c80-54d3-11eb-9013-915f50fa685a.png)

5. Enter this url, replacing the id below with the extracted id: `health-care/covid-19-vaccine/stay-informed/unsubscribe?sid=0FF5B689BADE119592161047302009611`

## Screenshots
Success:
![image](https://user-images.githubusercontent.com/2481110/104351148-072aa500-54d3-11eb-9577-db19073aaf2d.png)

Fail:
![image](https://user-images.githubusercontent.com/2481110/104350998-d0548f00-54d2-11eb-9557-d725a145e66f.png)

## Acceptance criteria
- [x] Associated BE endpoint is called when FE `unsubscribe` URL is accessed
- [x] Success or Failure verbiage is displayed to the user based on response from vets-api

## Definition of done
